presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/RuleAssert.java

```diff
@@ -106,6 +117,35 @@ public void matches(PlanMatchPattern pattern)
                     actual.getOutputSymbols()));
         }
 
-        assertPlan(session, metadata, new Plan(actual, symbolAllocator), pattern);
+        plan = actual;
+        symbols = symbolAllocator.getTypes();
+
+        return this;
+    }
+
+    public RuleAssert isFiredOnlyOnceAndThen()
+    {
+        return isFiredMultipleTimesAndThen(1);
+    }
+
+    public RuleAssert isFiredMultipleTimesAndThen(int count)
```

Comment:
I'm not 100% convinced we want to extend the framework to allow these kinds of iterative tests. We miss the ability to test the intermediate transformations, for instance. My original intent was for the rule assertion framework to verify single rule applications.

I do think it's important to test iterative application of rules, but that can be done with the logical planner tests (TestLogicalPlanner and friends), or by building something like it for sets of rules.